### ISSUE TITLE
Crash in InspectableWebContentsImpl::WebContentsDestroyed

### ISSUE THREAD

**Issue by CharlieHess** [electron/electron#9188]
* Electron version: 1.6.3
* Operating system: macOS 10.12.3

Been seeing this crash somewhat frequently when closing devTools on 1.6.x:

<details>

```
Process:               Electron [9829]
Path:                  /Users/USER/*/Electron.app/Contents/MacOS/Electron
Identifier:            com.github.electron
Version:               1.6.3 (1.6.3)
Code Type:             X86-64 (Native)
Parent Process:        node [9828]
Responsible:           Hyper [35816]
User ID:               502

Date/Time:             2017-04-13 14:46:40.858 -0700
OS Version:            Mac OS X 10.12.3 (16D32)
Report Version:        12
Anonymous UUID:        D97A7BE2-F8CD-6D3E-99FF-F7947FBB2647

Sleep/Wake UUID:       ABF9F3BE-62C1-4593-8BCF-C614B7C570E9

Time Awake Since Boot: 860000 seconds
Time Since Wake:       15000 seconds

System Integrity Protection: enabled

Crashed Thread:        0  CrBrowserMain  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       EXC_I386_GPFLT

Application Specific Information:
Performing @selector(_close:) from sender _NSThemeCloseWidget 0x7fc4f544d010

Thread 0 Crashed:: CrBrowserMain  Dispatch queue: com.apple.main-thread
0   com.github.electron.framework 	0x0000000104cfb2fc brightray::InspectableWebContentsImpl::WebContentsDestroyed() + 60 (inspectable_web_contents_impl.cc:652)
1   com.github.electron.framework 	0x0000000104cfb38d non-virtual thunk to brightray::InspectableWebContentsImpl::WebContentsDestroyed() + 13 (inspectable_web_contents_impl.cc:647)
2   com.github.electron.framework 	0x00000001053a1d11 content::WebContentsImpl::~WebContentsImpl() + 2465 (weak_ptr.h:225)
3   com.github.electron.framework 	0x00000001053a244e content::WebContentsImpl::~WebContentsImpl() + 14 (web_contents_impl.cc:483)
4   com.github.electron.framework 	0x0000000104cf949d brightray::InspectableWebContentsImpl::CloseDevTools() + 61 (memory:2761)
5   com.github.electron.framework 	0x0000000104cfd4fd -[BRYInspectableWebContentsView windowWillClose:] + 29 (bry_inspectable_web_contents_view.mm:224)
6   com.apple.CoreFoundation      	0x00007fff77017a6c __CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__ + 12
7   com.apple.CoreFoundation      	0x00007fff7701796b _CFXRegistrationPost + 427
8   com.apple.CoreFoundation      	0x00007fff770176d2 ___CFXNotificationPost_block_invoke + 50
9   com.apple.CoreFoundation      	0x00007fff76fd4d63 -[_CFXNotificationRegistrar find:object:observer:enumerator:] + 1827
10  com.apple.CoreFoundation      	0x00007fff76fd3d9c _CFXNotificationPost + 604
11  com.apple.Foundation          	0x00007fff789faa37 -[NSNotificationCenter postNotificationName:object:userInfo:] + 66
12  com.apple.AppKit              	0x00007fff74dca548 __18-[NSWindow _close]_block_invoke + 205
13  com.apple.AppKit              	0x00007fff74dca42c -[NSWindow _close] + 365
14  com.apple.AppKit              	0x00007fff74e59031 -[NSWindow __close] + 305
15  libsystem_trace.dylib         	0x00007fff8c78bc3d _os_activity_initiate + 61
16  com.apple.AppKit              	0x00007fff752afc9c -[NSApplication(NSResponder) sendAction:to:from:] + 456
17  com.apple.AppKit              	0x00007fff74d95460 -[NSControl sendAction:to:] + 86
18  com.apple.AppKit              	0x00007fff74d95388 __26-[NSCell _sendActionFrom:]_block_invoke + 136
19  libsystem_trace.dylib         	0x00007fff8c78bc3d _os_activity_initiate + 61
20  com.apple.AppKit              	0x00007fff74d952e0 -[NSCell _sendActionFrom:] + 128
21  com.apple.AppKit              	0x00007fff74dd7cd9 -[NSButtonCell _sendActionFrom:] + 98
22  libsystem_trace.dylib         	0x00007fff8c78bc3d _os_activity_initiate + 61
23  com.apple.AppKit              	0x00007fff74d93bc6 -[NSCell trackMouse:inRect:ofView:untilMouseUp:] + 2481
24  com.apple.AppKit              	0x00007fff74dd7a12 -[NSButtonCell trackMouse:inRect:ofView:untilMouseUp:] + 798
25  com.apple.AppKit              	0x00007fff74d9257b -[NSControl mouseDown:] + 832
26  com.apple.AppKit              	0x00007fff74e3246e -[_NSThemeWidget mouseDown:] + 87
27  com.apple.AppKit              	0x00007fff75427603 -[NSWindow(NSEventRouting) _handleMouseDownEvent:isDelayedEvent:] + 6341
28  com.apple.AppKit              	0x00007fff75423e20 -[NSWindow(NSEventRouting) _reallySendEvent:isDelayedEvent:] + 1942
29  com.apple.AppKit              	0x00007fff754232be -[NSWindow(NSEventRouting) sendEvent:] + 541
30  com.github.electron.framework 	0x0000000104cfdab4 -[EventDispatchingWindow sendEvent:] + 52 (event_dispatching_window.mm:11)
31  com.apple.AppKit              	0x00007fff752abbf5 -[NSApplication(NSEvent) sendEvent:] + 1145
32  com.github.electron.framework 	0x0000000104c3fdcb -[AtomApplication sendEvent:] + 75 (atom_application.mm:24)
33  com.apple.AppKit              	0x00007fff74b27f81 -[NSApplication run] + 1002
34  com.github.electron.framework 	0x0000000104dcaf0e base::MessagePumpNSApplication::DoRun(base::MessagePump::Delegate*) + 334 (message_pump_mac.mm:654)
35  com.github.electron.framework 	0x0000000104dca57c base::MessagePumpCFRunLoopBase::Run(base::MessagePump::Delegate*) + 92 (message_pump_mac.mm:218)
36  com.github.electron.framework 	0x0000000104dc6abe base::MessageLoop::RunHandler() + 94 (message_loop.cc:379)
37  com.github.electron.framework 	0x0000000104ddb613 base::RunLoop::Run() + 115 (run_loop.cc:36)
38  com.github.electron.framework 	0x000000010502b31b content::BrowserMainLoop::MainMessageLoopRun() + 155 (browser_main_loop.cc:1500)
39  com.github.electron.framework 	0x000000010502b170 content::BrowserMainLoop::RunMainMessageLoopParts() + 64 (atomic:890)
40  com.github.electron.framework 	0x000000010502d8b2 content::BrowserMainRunnerImpl::Run() + 18 (memory:2761)
41  com.github.electron.framework 	0x000000010502724c content::BrowserMain(content::MainFunctionParams const&) + 124 (browser_main.cc:46)
42  com.github.electron.framework 	0x0000000104f8ce6d content::ContentMainRunnerImpl::Run() + 477 (content_main_runner.cc:774)
43  com.github.electron.framework 	0x0000000104f8bf66 content::ContentMain(content::ContentMainParams const&) + 54 (content_main.cc:20)
44  com.github.electron.framework 	0x0000000104ba4687 AtomMain + 71 (atom_library_main.mm:24)
45  com.github.electron           	0x0000000104b9cf26 main + 38 (atom_main.cc:143)
46  libdyld.dylib                 	0x00007fff8c559255 start + 1
```

</details>

### How to reproduce

We have a developer shortcut that iterates through all `webContents` in the main process and opens or closes devTools for some subset of them. Typically after toggling this a few times, if I close a devTools window in the regular fashion (hitting the `x`), this crash occurs.

**Comment by deepak1556** [comment]
@CharlieHess can you check if the crash happens with 1.6.5-beta ? Also are you able to reproduce with a minimal example ?

**Comment by michaelenger** [comment]
I've experienced the same crashes with the same error log, but it was solved after I updated to 1.6.5

**Comment by CharlieHess** [comment]
I haven't been able to reproduce on 1.6.5. Thanks for the info.
